batch/sc-batch-fargate.yaml

```diff
@@ -0,0 +1,200 @@
+  Schedule:
+    Description: >
+      Schedule to execute the docker, can be a rate or a cron schedule. Format at
+      https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+    Type: String
+    Default: cron(0 09 ? * MON *)  # Run every Monday at 9am
+    ConstraintDescription: "Use schedule format: https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html"
+  SynapsePAT:
+    Type: String
+    NoEcho: true
+    Default: ""
+    Description: >
+      (Optional) The Synapse personal access token (PAT).
+      https://docs.synapse.org/rest/POST/personalAccessToken.html
+Resources:
+  ComputeEnvironment:
+    Type: AWS::Batch::ComputeEnvironment
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3008
+    Properties:
+      Type: MANAGED
+      State: ENABLED
+      ServiceRole: !Ref ServiceRole
+      ComputeResources:
+        Type: FARGATE
+        MaxvCpus: 1
+        Subnets:
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet1
+          - !ImportValue us-east-1-scriptrunnervpc-PrivateSubnet2
+        SecurityGroupIds:
+          - !Ref SecurityGroup
+  ServiceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: 'Allow'
+            Principal:
+              Service:
+                - 'batch.amazonaws.com'
+            Action:
+              - 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - 'arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole'
+  JobQueue:
+    Type: AWS::Batch::JobQueue
+    Properties:
+      ComputeEnvironmentOrder:
+        - ComputeEnvironment: !Ref ComputeEnvironment
+          Order: 1
+      Priority: 1
+      State: ENABLED
+  SchedulingPolicy:
+    Type: AWS::Batch::SchedulingPolicy
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3001
+    Properties:
+      FairsharePolicy:
+        ShareDecaySeconds: 5
+        ComputeReservation: 2
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      VpcId: !ImportValue us-east-1-scriptrunnervpc-VPCId
+      GroupDescription: !Sub 'SC for ${AWS::StackName}}'
+  JobDefinition:
+    Type: AWS::Batch::JobDefinition
+    Properties:
+      Type: container
+      JobDefinitionName: { Ref: "AWS::StackName" }
+      PropagateTags: True
+      PlatformCapabilities:
+        - FARGATE
+      Timeout:
+        AttemptDurationSeconds: 600
+      RetryStrategy:
+        Attempts: 1
+      ContainerProperties:
+        Command: !Ref Command
+        Image: !Ref Image
+        Environment:
+          - Name: SynapsePAT
+            Value: !Ref SynapsePAT
+        NetworkConfiguration:
+          AssignPublicIp: ENABLED
+        ResourceRequirements:
+          - Type: VCPU
+            Value: 0.5
+          - Type: MEMORY
+            Value: 1024
```

Comment:
I think its fair to assume for most jobs that it won't take much CPU and memory, but maybe this should be configurable as well.

Reply:
Setup a mapping to allow users to select memory options that will map to CPU shares.  This will allow users to configure the memory for the container but will also constrain CPU shares to a matching pair of memory and VCPU shares.